### PR TITLE
[kube-prometheus-stack] fix reloader-web's app protocol

### PR DESCRIPTION
**Pull request by abaguas**
#### What this PR does / why we need it

To integrate this port with applications such as istio the applicationProtocol should be set explicitly.
One solution to enable this integration is to add an `appProtocol` attribute to the port as explained in the [Kubernetes documentation](https://kubernetes.io/docs/concepts/services-networking/service/#application-protocol) and implemented in this PR. Another alternative is to prefix the port with `http-`.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by abaguas** [comment]
If you prefered I can rename the port to `http-reloader-web` or add the `appProtocol` to the other `http` ports

**Comment by GMartinez-Sisti** [comment]
Hi @abaguas! Thank you for the contribution 🙏 

Looks like the `appProtocol` feature graduated on K8S `1.20`, since we support `1.19` it would be nice to have some validation before adding it. Maybe something like this to append it conditionally:

https://github.com/prometheus-community/helm-charts/blob/3352ef86286a7e8f07266ff3da1a575aa7368c8f/charts/kube-prometheus-stack/templates/_helpers.tpl#L170-L179

WDYT?

**Comment by abaguas** [comment]
Makes sense, thank you for the tip. I implemented it and tested with

```
$ helm package . && helm template kube-prometheus-stack kube-prometheus-stack-50.3.2.tgz -n monitoring --set kubeTargetVersionOverride="1.19.0" > manifests.yaml      
$ helm package . && helm template kube-prometheus-stack kube-prometheus-stack-50.3.2.tgz -n monitoring --set kubeTargetVersionOverride="1.20.0" > manifests-after.yaml
$ diff manifests.yaml manifests-after.yaml 
1389a1390
>     appProtocol: http
1584a1586
>     appProtocol: http
```

File: charts/kube-prometheus-stack/templates/prometheus/service.yaml [hunk 1]
```diff
@@ -43,6 +43,10 @@ spec:
     port: {{ .Values.prometheus.service.port }}
     targetPort: {{ .Values.prometheus.service.targetPort }}
   - name: reloader-web
+    {{- $kubeTargetVersion := default .Capabilities.KubeVersion.GitVersion .Values.kubeTargetVersionOverride }}
```
Comment (QuentinBisson) [inline charts/kube-prometheus-stack/templates/prometheus/service.yaml:46]: Could you move this at the top?

Comment (QuentinBisson) [reply charts/kube-prometheus-stack/templates/prometheus/service.yaml:46]: Otherwise looks good

**Review by QuentinBisson — DISMISSED** [review]

**Comment by QuentinBisson** [comment]
@abaguas Can you sign your commits?

**Review by QuentinBisson — APPROVED** [review]

**Pull request merged**